### PR TITLE
HPCC-30087 Generate python310.zip stdlib file

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes eclcc issues when the pyembed version and local python versions don't
match.

Constant folded python code should now work without a local python environment.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@ghalliday - FYI the official python310.zip file only exists in the windows embedded download, if we want to take advantage of it in non windows builds, then manually recreating it is simpler/preferable?

File: plugins/py3embed/CMakeLists.txt [hunk 1]
```diff
@@ -53,6 +57,44 @@ elseif(NOT APPLE)
     set_target_properties(py3embed PROPERTIES NO_SONAME 1)
 endif()
 
+if (CREATE_STDLIB_ZIP)
+    set(_py3_zip_temp_dir "${CMAKE_CURRENT_BINARY_DIR}/python_embeddable_zip_temp")
+    set(_py3_zip_path "${_py3_zip_temp_dir}/python${Python3_VERSION_MAJOR}${Python3_VERSION_MINOR}.zip")
```
Comment (ghalliday) [inline plugins/py3embed/CMakeLists.txt:62]: Is this name correct for debug versions?  (If so it is only a problem for developers, so could be a separate PR)

**Review by ghalliday — APPROVED** [review]
@GordonSmith Is there a separate JIRA to modify the vcpkg so it becomes a build artefact?  This looks like a temporary work-around.

File: plugins/py3embed/CMakeLists.txt (hunk 1 above)
Comment (GordonSmith) [reply plugins/py3embed/CMakeLists.txt:62]: I think this is still valid - I will recheck now...

Comment (GordonSmith) [reply plugins/py3embed/CMakeLists.txt]: It was still valid for the "Lib" folder, but not for the zip option.  Fixed...

**Review by ghalliday — APPROVED** [review]
@GordonSmith please squash
